config/sections/pages.php

```diff
@@ -232,11 +232,30 @@
 				return false;
 			}
 
-			if (in_array($this->status, ['draft', 'all']) === false) {
+			if ($this->isFull() === true) {
 				return false;
 			}
 
-			if ($this->isFull() === true) {
+			// ensure that the add button is only shown when
+			// the section is displaying pages with the same status
+			// as the status that is used for page creation
+			$statuses = [];
+
+			foreach ($this->blueprintNames() as $blueprint) {
+				try {
+					$props      = Blueprint::load('pages/' . $blueprint);
+					$statuses[] = $props['create']['status'] ?? 'draft';
+				} catch (Throwable) {
+				}
+			}
+
+			$statuses = array_unique($statuses);
+
+			if (count($statuses) > 1 && $this->status !== 'all') {
+				return false;
+			}
+
+			if (in_array($this->status, [...$statuses, 'all']) === false) {
 				return false;
 			}
```

Comment:
Since `$this->status === 'all'` always allows the add button, we could have this as a shortcut above the new logic:

```php
if ($this->status === 'all') {
	return true;
}
```

These checks could then be:

```suggestion
			if (count($statuses) > 1 || $this->status !== $statuses[0]) {
				return false;
			}
```

Reply:
Good call.

I think the last check has to be a bit different as the array can also be empty.

Reply:
I guess it would be `count($statuses) !== 1` then, right?